packages/time-picker/test/form-input.test.js

```diff
@@ -49,159 +49,341 @@ describe('form input', () => {
       timePicker = fixtureSync(`<vaadin-time-picker></vaadin-time-picker>`);
-      comboBox = timePicker.$.comboBox;
-      inputElement = timePicker.inputElement;
+      validateSpy = sinon.spy(timePicker, 'validate');
     });
 
-    it('should have synced invalid property with input on validation with required flag', () => {
-      timePicker.name = 'foo';
-      timePicker.required = true;
-      timePicker.min = '14:00';
-      timePicker.value = '13:00';
-      comboBox.dispatchEvent(new CustomEvent('change', { bubbles: true }));
+    it('should pass validation by default', () => {
+      expect(timePicker.checkValidity()).to.be.true;
+      expect(timePicker.validate()).to.be.true;
+    });
 
-      expect(inputElement.hasAttribute('invalid')).to.be.true;
+    it('should not validate on value input', () => {
+      setInputValue(timePicker, '12:00');
+      expect(validateSpy.calledOnce).to.be.false;
+    });
 
-      timePicker.value = '12:00';
-      expect(inputElement.hasAttribute('invalid')).to.be.true;
+    it('should validate on value commit', () => {
+      setInputValue(timePicker, '12:00');
+      enter(timePicker.inputElement);
+      expect(validateSpy.calledOnce).to.be.true;
     });
 
-    it('should validate correctly with required flag', () => {
-      timePicker.name = 'foo';
-      timePicker.required = true;
+    it('should validate on input focusout', () => {
+      focusout(timePicker.inputElement);
+      expect(validateSpy.calledOnce).to.be.true;
+    });
 
-      expect(timePicker.validate()).to.equal(false);
-      expect(timePicker.invalid).to.be.equal(true);
+    it('should be possible to force invalid status', () => {
+      timePicker.invalid = true;
+      expect(inputElement.hasAttribute('invalid')).to.be.true;
+    });
 
-      timePicker.value = '13:00';
-      expect(timePicker.validate()).to.equal(true);
-      expect(timePicker.invalid).to.be.equal(false);
+    it('should fire a validated event on validation success', () => {
+      const validatedSpy = sinon.spy();
+      timePicker.addEventListener('validated', validatedSpy);
+      timePicker.validate();
+
+      expect(validatedSpy.calledOnce).to.be.true;
+      const event = validatedSpy.firstCall.args[0];
+      expect(event.detail.valid).to.be.true;
     });
 
-    it('should validate correctly with required flag on focusout', () => {
-      timePicker.name = 'foo';
+    it('should fire a validated event on validation failure', () => {
+      const validatedSpy = sinon.spy();
+      timePicker.addEventListener('validated', validatedSpy);
       timePicker.required = true;
+      timePicker.validate();
 
-      focusout(inputElement);
-      expect(timePicker.invalid).to.be.equal(true);
+      expect(validatedSpy.calledOnce).to.be.true;
+      const event = validatedSpy.firstCall.args[0];
+      expect(event.detail.valid).to.be.false;
     });
+  });
 
-    it('should respect min value during validation', () => {
-      timePicker.step = '0.5';
```

Comment:
**note:** `step` doesn't seem to participate in the validation, so I don't use it in the new version of tests.